src/core/qgsmaplayer.h

```diff
@@ -528,6 +528,96 @@ class CORE_EXPORT QgsMapLayer : public QObject
     static QString capitalizeLayerName( const QString &name );
 
     /**
+     * Retrieve the metadata URI for this layer
+     * (either as a .qmd file on disk or as a
+     * record in the users style table in their personal qgis.db)
+     * \returns a QString with the metadata file name
+     * \since QGIS 3.0
+     */
+    virtual QString metadataURI() const;
+
+    /**
+     * Export the properties of this layer as named metadata in a QDomDocument
+     * \param doc the target QDomDocument
+     * \param errorMsg this QString will be initialized on error
+     * \since QGIS 3.0
+     */
+    void exportNamedMetadata( QDomDocument &doc, QString &errorMsg ) const;
```

Comment:
To me this method would belong better in QgsLayerMetadata itself (same with saveNamedMetadata, loadNamedMetadata, loadNamedMetadataFromDatabase, importNamedMetadata), leaving only the default related methods in QgsMapLayer

Reply:
Do you mean that I can still let the main code for dealing with the database and the flag in QgsMapLayer and in QgsLayerMetadata I can just call the function from QgsMapLayer with the correct flag?
I did that to avoid duplication between these two features which are very close (save style and save metadata)